### PR TITLE
Add gasPrice param to add wallet

### PR DESCRIPTION
**Pull request by wesgraham**
Adds gasPrice param to add wallet
Ticket: [BG-23720](https://bitgoinc.atlassian.net/browse/BG-23720)

### CODE REVIEW

File: modules/core/test/v2/unit/wallets.ts [hunk 1]
```diff
@@ -94,6 +98,22 @@ describe('V2 Wallets:', function() {
...
+      const ethBitGo = new TestBitGo({ env: 'mock' });
+      ethBitGo.initializeTestVars();
+      const ethWallets = ethBitGo.coin('teth').wallets();
+      nock(bgUrl)
+        .post('/api/v2/teth/wallet', function(body) {
+          body.type.should.equal('custodial');
+          body.should.not.have.property('keys');
+          body.should.not.have.property('m');
+          body.should.not.have.property('n');
+          return true;
```
Comment (marktoda) [inline modules/core/test/v2/unit/wallets.ts:112]: These tests are intended to check that your custom parameters are being passed correctly. To do so, add a check in this callback to ensure that the custom gas price is passed as intended

File: modules/core/test/v2/unit/wallets.ts [hunk 2]
```diff
@@ -94,6 +98,23 @@ describe('V2 Wallets:', function() {
...
     }));
+
+    it('creates a wallet with custom gasPrice', co(function *createWalletCustomGasPrice() {
+      const ethBitGo = new TestBitGo({ env: 'mock' });
+      ethBitGo.initializeTestVars();
+      const ethWallets = ethBitGo.coin('teth').wallets();
+      nock(bgUrl)
+        .post('/api/v2/teth/wallet', function(body) {
+          body.type.should.equal('custodial');
+          body.gasPrice.should.equal(20000000000);
```
Comment (twtaylorbitgo) [inline modules/core/test/v2/unit/wallets.ts:109]: question: are you sure gasPrice isn't coming back as a string? most of our stuff is BigNumber at the API level

File: modules/core/test/v2/unit/wallets.ts [hunk 3]
```diff
@@ -94,6 +98,23 @@ describe('V2 Wallets:', function() {
...
+          body.should.not.have.property('keys');
+          body.should.not.have.property('m');
+          body.should.not.have.property('n');
+          return true;
+        })
+        .reply(200, {});
+      yield ethWallets.add({ label: 'label', enterprise: 'enterprise', type: 'custodial', gasPrice: 20000000000 });
```
Comment (twtaylorbitgo) [inline modules/core/test/v2/unit/wallets.ts:116]: this might need to be a string if it's in WEI

File: modules/core/test/v2/unit/wallets.ts (hunk 1 above)
Comment (tylerlevine) [reply modules/core/test/v2/unit/wallets.ts:113]: There's actually a slightly cleaner way to do this - set up your nock so it only matches if the body params are as you expect.

This isn't incorrect though I'm so not requesting changes for it.

**Pull request merged**